vendor-asset_request-apiary-scenario

martinconstante opened the pull request:
This PR contains the vendor asset request apiary scenario that consists of the creation of the tenant on the Vendor System and when the tenant is in status "ready", accept the PR into Cloudblue Connect.

marcserrat commented on examples/apiary_scenarios/asset_accept_workflow.js:
```diff
@@ -0,0 +1,95 @@
+
+const client = new ConnectClient(
+  'https://api.connect.cloudblue.com/public/v1',
+  'ApiKey SU-069-215-193:e3ae33ea2a7ec62b434452d4feb92eb32dbf57d3',
```
no keys please

marcserrat commented on examples/apiary_scenarios/asset_accept_workflow.js:
```diff
@@ -0,0 +1,95 @@
+);
+const fulfillment = new Fulfillment(client);
+const inventory = new Inventory(client);
+const urlBase = 'https://private-368580-vendorexample.apiary-mock.com';
```
in python example we set to change me or similar

marcserrat commented on examples/apiary_scenarios/asset_accept_workflow.js:
```diff
@@ -0,0 +1,95 @@
+const urlBase = 'https://private-368580-vendorexample.apiary-mock.com';
+
+/**
+ * Accept Purchase Request into Cloudblue Connect
```
process Purchase request from CloudBlue Connect will be better no?

marcserrat commented on examples/apiary_scenarios/asset_accept_workflow.js:
```diff
@@ -0,0 +1,95 @@
+ *
+ * @returns {Array}    An array of Template objects.
+ */
+function acceptTenant(element) {
```
unsure why acceptTenant, is not processRequest or similar?

marcserrat commented on examples/apiary_scenarios/asset_request_workflow.js:
```diff
@@ -0,0 +1,109 @@
+
+const client = new ConnectClient(
+  'https://api.connect.cloudblue.com/public/v1',
+  'ApiKey SU-069-215-193:e3ae33ea2a7ec62b434452d4feb92eb32dbf57d3',
```
shall not be here, also since is used in multiple places, better to move to a configuration file?

marcserrat commented on examples/apiary_scenarios/asset_request_workflow.js:
```diff
@@ -0,0 +1,109 @@
+ */
+function checkTenant(requests) {
+  requests.forEach((element) => {
+    // Verificamos si existe el tennant
```
spanish

marcserrat commented on examples/apiary_scenarios/yarn.lock:
```diff
@@ -0,0 +1,21 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
shall be file commited?

martinconstante commented:
________________________________

In examples/apiary_scenarios/asset_accept_workflow.js<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_cloudblue_connect-2Djavascript-2Dsdk_pull_20-23discussion-5Fr428505573&d=DwMCaQ&c=--1RjWWBW4Kf6aBAaj53vPItwfT0BR1YjSDV46P5EvE&r=Zd7qmpHgLIqStNeYcAmkS_Bv1HWMXa8wq_JrGguVNiE&m=Ft-e22_CN6oWWB5Zs-gyKg01sTAXC7Rj1knoPorIENQ&s=zv3TG38rnx-UYUHPdPvwuuGe64qPe1BIjHoI2y02_Ps&e=>:

> + * example of API Vendor System implemented in apiary.io

+ * and documented in the Cloudblue Connect Documentation Portal:

+ * https://connect.cloudblue.com/community/sdk/vendor-scenario-example/fulfillment/

+ *

+ * This module accept the Purchase Request into Connect when the tenant is ready in Vendor System.

+ *

+ * @category Operations

+ */

+

+const client = new ConnectClient(

+  'https://api.connect.cloudblue.com/public/v1',

+  'ApiKey SU-069-215-193:e3ae33ea2a7ec62b434452d4feb92eb32dbf57d3',

+);

+const fulfillment = new Fulfillment(client);

+const inventory = new Inventory(client);

+const urlBase = 'https://private-368580-vendorexample.apiary-mock.com';

in python example we set to change me or similar
Changed.

________________________________

In examples/apiary_scenarios/asset_accept_workflow.js<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_cloudblue_connect-2Djavascript-2Dsdk_pull_20-23discussion-5Fr428505879&d=DwMCaQ&c=--1RjWWBW4Kf6aBAaj53vPItwfT0BR1YjSDV46P5EvE&r=Zd7qmpHgLIqStNeYcAmkS_Bv1HWMXa8wq_JrGguVNiE&m=Ft-e22_CN6oWWB5Zs-gyKg01sTAXC7Rj1knoPorIENQ&s=yMo2KiOScqOuEHRuBn4vbhMpify8WLCnJNb4ZHtk88M&e=>:

> + *

+ * This module accept the Purchase Request into Connect when the tenant is ready in Vendor System.

+ *

+ * @category Operations

+ */

+

+const client = new ConnectClient(

+  'https://api.connect.cloudblue.com/public/v1',

+  'ApiKey SU-069-215-193:e3ae33ea2a7ec62b434452d4feb92eb32dbf57d3',

+);

+const fulfillment = new Fulfillment(client);

+const inventory = new Inventory(client);

+const urlBase = 'https://private-368580-vendorexample.apiary-mock.com';

+

+/**

+ * Accept Purchase Request into Cloudblue Connect

process Purchase request from CloudBlue Connect will be better no?
Changed.

________________________________

In examples/apiary_scenarios/asset_accept_workflow.js<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_cloudblue_connect-2Djavascript-2Dsdk_pull_20-23discussion-5Fr428506354&d=DwMCaQ&c=--1RjWWBW4Kf6aBAaj53vPItwfT0BR1YjSDV46P5EvE&r=Zd7qmpHgLIqStNeYcAmkS_Bv1HWMXa8wq_JrGguVNiE&m=Ft-e22_CN6oWWB5Zs-gyKg01sTAXC7Rj1knoPorIENQ&s=xh710XHo-NlnXDFF3zAWItZKiypKwnzxI8rzhgIQSxg&e=>:

> +const client = new ConnectClient(

+  'https://api.connect.cloudblue.com/public/v1',

+  'ApiKey SU-069-215-193:e3ae33ea2a7ec62b434452d4feb92eb32dbf57d3',

+);

+const fulfillment = new Fulfillment(client);

+const inventory = new Inventory(client);

+const urlBase = 'https://private-368580-vendorexample.apiary-mock.com';

+

+/**

+ * Accept Purchase Request into Cloudblue Connect

+ *

+ * @param   {object}  element  The request of Cloudblue Connect.

+ *

+ * @returns {Array}    An array of Template objects.

+ */

+function acceptTenant(element) {

unsure why acceptTenant, is not processRequest or similar?
Changed.



________________________________

In examples/apiary_scenarios/asset_request_workflow.js<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_cloudblue_connect-2Djavascript-2Dsdk_pull_20-23discussion-5Fr428507530&d=DwMCaQ&c=--1RjWWBW4Kf6aBAaj53vPItwfT0BR1YjSDV46P5EvE&r=Zd7qmpHgLIqStNeYcAmkS_Bv1HWMXa8wq_JrGguVNiE&m=Ft-e22_CN6oWWB5Zs-gyKg01sTAXC7Rj1knoPorIENQ&s=IPy4QU9PN3IiR8wnr2JFUlDeBNnN0n82o1VXurbf7lg&e=>:

> +/**

+ * This module is an example of fulfillment flow between

+ * Cloudblue Connect and Vendor System.

+ * This connect with the Vendor System using an

+ * example of API Vendor System implemented in apiary.io

+ * and documented in the Cloudblue Connect Documentation Portal:

+ * https://connect.cloudblue.com/community/sdk/vendor-scenario-example/fulfillment/

+ *

+ * This module creates a tenant in the Vendor System when is created in Cloudblue Connect.

+ *

+ * @category Operations

+ */

+

+const client = new ConnectClient(

+  'https://api.connect.cloudblue.com/public/v1',

+  'ApiKey SU-069-215-193:e3ae33ea2a7ec62b434452d4feb92eb32dbf57d3',

shall not be here, also since is used in multiple places, better to move to a configuration file?
Changed.



________________________________

In examples/apiary_scenarios/asset_request_workflow.js<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_cloudblue_connect-2Djavascript-2Dsdk_pull_20-23discussion-5Fr428507733&d=DwMCaQ&c=--1RjWWBW4Kf6aBAaj53vPItwfT0BR1YjSDV46P5EvE&r=Zd7qmpHgLIqStNeYcAmkS_Bv1HWMXa8wq_JrGguVNiE&m=Ft-e22_CN6oWWB5Zs-gyKg01sTAXC7Rj1knoPorIENQ&s=xMdyJzi5kbQ5SQQURvuU9Cc9aAUKnVrXbckcNqi4eJk&e=>:

> +    cache: 'no-cache',

+  })

+    .then((response) => response.json())

+    .then((data) => console.log('data = ', data))

+    .catch((err) => console.error(err));

+}

+

+/**

+ * Check if the request exist into Vendor Portal

+ *

+ * @param   {object}  requests  The request of Cloudblue Connect.

+ *

+ */

+function checkTenant(requests) {

+  requests.forEach((element) => {

+    // Verificamos si existe el tennant

Spanish
Changed.



________________________________

In examples/apiary_scenarios/yarn.lock<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_cloudblue_connect-2Djavascript-2Dsdk_pull_20-23discussion-5Fr428508052&d=DwMCaQ&c=--1RjWWBW4Kf6aBAaj53vPItwfT0BR1YjSDV46P5EvE&r=Zd7qmpHgLIqStNeYcAmkS_Bv1HWMXa8wq_JrGguVNiE&m=Ft-e22_CN6oWWB5Zs-gyKg01sTAXC7Rj1knoPorIENQ&s=p0tnU5gYipe0EoRyPhaIcfDCh519UUU4WbYn09UqWQc&e=>:

> @@ -0,0 +1,21 @@

+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.

shall be file commited?

Yes, It must be commited.
https://classic.yarnpkg.com/en/docs/migrating-from-npm

[cid:image001.jpg@01D62F64.F750EFC0]

La información contenida en este mensaje es confidencial. En caso de que reciba este mensaje por error le rogamos lo comunique a la mayor brevedad al emisor y proceda a su eliminación definitiva, absteniéndose de copiar, almacenar o difundir su contenido. De acuerdo con lo establecido en la Ley Orgánica 15/1999, de Protección de Datos de Carácter Personal y en el Reglamento de Desarrollo 1720/2007, los datos personales que facilite a través de la dirección de correo indicada serán incorporados a un fichero titularidad de INGRAM MICRO, S.L.U., con domicilio en C/ Antonio Machado, 78-80 1ª y 2ª pl. Business Park ( 08840-Viladecans). Mediante el envío de sus datos, Ud. otorga su consentimiento expreso a INGRAM MICRO, S.L.U, para el tratamiento de sus datos, con la finalidad de atender a su consulta y/o mantener la relación profesional, comercial, y/o contractual que en su caso establezca con INGRAM MICRO, S.L.U. Puede ejercitar sus derechos de acceso, rectificación, cancelación y oposición notificándolo por escrito a la dirección del remitente, o a la siguiente dirección de correo nuevascuentas@ingrammicro.es. De acuerdo con la Ley 34/2002, de Servicios de la Sociedad de la Información y de Comercio Electrónico, Vd. podrá oponerse en cualquier momento al tratamiento de sus datos con fines promocionales notificándonoslo por escrito a la dirección de correo mencionada.
.................................................................................................................................................................................................................................................
The information contained in this message is confidential. If you receive this message by error please notify it as soon as possible to the sender and proceed to their final elimination by not copy, store or distribute its content. In accordance of what is stated in the Law 15/1999, of Data Personal Protection and Regulation Rule 1720/2007, the personal data provided through the email address you entered will be included in a file owned by INGRAM MICRO, SLU, located at C/ Antonio Machado, 78-80 1ª y 2ª pl. Business Park ( 08840-Viladecans). By submitting your data, you expressly give your consent to INGRAM MICRO, SLU, to the treatment of your data, in order to answer to your questions and / or keep the professional, commercial relationship  and / or contractual set with INGRAM MICRO, SLU You can exercise your rights of access, rectification, cancellation and opposition by giving written notification to the sender address or to  the following email:  nuevascuentas@ingrammicro.es. According to Law 34/2002, of the Information Society and Electronic Commerce, you may object at any time to your data treatment for promotional purposes by notifying us in writing to the email address above.
[Ingram_2818e5de]
